api/server/services/AuthService.js

```diff
@@ -188,9 +186,56 @@ const resetPassword = async (userId, token, password) => {
   return { message: 'Password reset was successful' };
 };
 
+/**
+ * Set Auth Tokens
+ *
+ * @param {String} userId
+ * @param {Object} res
+ * @param {String} sessionId
+ * @returns
+ */
+const setAuthTokens = async (userId, res, sessionId = null) => {
+  try {
+    const user = await User.findOne({ _id: userId });
+    const token = await user.generateToken();
+
+    let session;
+    let refreshTokenExpires;
+    if (sessionId) {
+      session = await Session.findById(sessionId);
+      refreshTokenExpires = session.expiration.getTime();
+    } else {
+      session = new Session({ user: userId });
+      refreshTokenExpires = Date.now() + eval(process.env.REFRESH_TOKEN_EXPIRY);
+    }
+    
+    const tokenExpires = eval(process.env.SESSION_EXPIRY);
+    const refreshToken = await session.generateRefreshToken();
+    
+    res.cookie('token', token, {
```

Comment:
We should no longer be setting access token in cookie. The advantage and reasoning for refresh token is that you don't have to set the access token in a cookie which is less secure. Instead, the api can response with a json object that has the access token, as well as setting the refresh token to the cookie. Then, the client takes the json access token and sets that to the axios authorization header. When the access token expires or the authorization header is lost due to something like refreshing the page, the refresh token is used to get a new access token. You're already doing most of this, so it should just be a small change. If you're unsure about any of this you can add me as a collaborator to your fork and I can help you out.

Reply:
Is the access token inside the refresh token?

Reply:
I just made the changes that removed setting the access token in a cookie and only keeps it as a value in the http header.  I am running some additional tests now to confirm the various login/logout/token expiration/refresh token expiration scenarios

Reply:
I saw somewhere that they were having problems with the nodejs distribution site, so I would wait a little bit and try running them again.

Reply:
So I removed the refreshTokenValidator because I think the silentRefresh fixed it.  I also remove the tokenExpires variable that was being used for the token cookie which is not longer needed. Feels nice to delete unused code.  However, I think it is working now because a refreshToken is being set on oauth authentication and the the silentRefresh is running at login which sets the corrrect auth context.  I do not know why login is not setting the correct auth context on login.

Reply:
I also did add you as a collaborator. I hope I did it right.

Reply:
I broke the registration components originally, now I think I have fixed it. Both Login and registration seems good.  However, social logins still seem to still do the blink login page before entering.  Technically it is all working without an access token cookie!

Reply:
Fixed the logout issue. I think the reload immediately following logout would at times interfere with the logout function.  This was actually impacting Chrome and Firefox, but you would only see it in the firefox system of not logging out.  However, Chrome would not always delete the refreshToken.  I also think a silentRefresh would run on the page reload causing the user to stay logged in by getting an up to date token. The logout seems to function without the reload.  Now the logout functions correctly on every call and removes the refreshToken everytime.

  const handleLogout = () => {
    logout();
    window.location.reload();
  };

Changed to:
    logout();